src/compiler/checker.ts

```diff
@@ -19877,7 +19877,8 @@ namespace ts {
             if (produceDiagnostics && targetType !== errorType) {
                 const widenedType = getWidenedType(exprType);
                 if (!isTypeComparableTo(targetType, widenedType)) {
-                    checkTypeComparableTo(exprType, targetType, errNode, Diagnostics.Type_0_cannot_be_converted_to_type_1);
+                    checkTypeComparableTo(exprType, targetType, errNode,
+                        Diagnostics.Type_0_cannot_be_converted_directly_to_type_1_because_it_is_neither_a_supertype_nor_a_subtype_of_1_If_this_is_intentional_convert_to_unknown_first);

```

Comment:
It's really not only about supertype vs. subtype: we have these relationships for subtyping, assignability, and comparability). Comparability has some other properties that are technically more lax.

> The comparability and assignment compatibility rules differ only in that
>
> * a union type is comparable to a target type if any of its constituents, rather than all of its constituents, are comparable to that target type,
> * whether a type is considered weak ([3.11.7](#3.11.7)) has no bearing on whether two types are comparable,
> * whether an object type has excess properties ([3.11.8](#3.11.8)) has no bearing on whether two types are comparable,
> * an object type with some optional property is comparable to an object type containing a property of the same name if it has a compatible type, regardless of whether that property is optional, and
> * when relating any two signatures, each signature is always instantiated using type Any for all type arguments.

So maybe you want something like

`Type '{0}' cannot be converted to type '{1}' because neither type sufficiently overlaps with the other.`

Reply:
@DanielRosenwasser worth reading the discussion in the linked issue IMO

Reply:
> The comparability and assignment compatibility rules differ only in that [...]

Where is this secret more-up-to-date version of the language spec?  [Aha](https://github.com/Microsoft/TypeScript/pull/17215/files#diff-a37bdfbd0d2f3627958003f87033871eR2393), web search is my friend.

Reply:
To try to move this forward, I went with the "neither type sufficiently overlaps with the other" wording.